app/screens/settings/notification_mention/notification_mention.tsx

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useReducer, useState} from 'react';
+import {Alert, ScrollView, Text, View} from 'react-native';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import FormattedText from '@components/formatted_text';
+import Section from '@components/section';
+import SectionItem from '@components/section_item';
+import {useTheme} from '@context/theme';
+import {t} from '@i18n';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type UserModel from '@typings/database/models/servers/user';
+
+const UPDATE_MENTION_PREF = 'UPDATE_MENTION_PREF';
+const INITIAL_STATE = {
+    firstName: false,
+    usernameMention: false,
+    channel: false,
+};
+type Action = {
+    type: string;
+    data: Partial<typeof INITIAL_STATE>;
+}
+const reducer = (state: typeof INITIAL_STATE, action: Action) => {
+    switch (action.type) {
+        case UPDATE_MENTION_PREF:
+            return {
+                ...state,
+                ...action.data,
+            };
+
+        default:
+            return state;
+    }
+};
+
+const mentionHeaderText = {
+    id: t('notification_settings.mentions.wordsTrigger'),
+    defaultMessage: 'WORDS THAT TRIGGER MENTIONS',
+};
+const replyHeaderText = {
+    id: t('notification_settings.mention.reply'),
+    defaultMessage: 'SEND REPLY NOTIFICATIONS FOR',
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+        },
+        input: {
+            color: theme.centerChannelColor,
+            fontSize: 12,
+            height: 40,
+        },
+        separator: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            flex: 1,
+            height: 1,
+            marginLeft: 15,
+        },
+        scrollView: {
+            flex: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.06),
+        },
+        scrollViewContent: {
+            paddingVertical: 35,
+        },
+        area: {
+            paddingHorizontal: 16,
+        },
+    };
+});
+
+type NotificationMentionProps = {
+    isCRTEnabled?: boolean;
+    currentUser?: UserModel;
+    mentionKeys: string;
+}
+const NotificationMention = ({currentUser, mentionKeys, isCRTEnabled}: NotificationMentionProps) => {
+    const [{firstName, usernameMention, channel}, dispatch] = useReducer(reducer, INITIAL_STATE);
+    const [replyNotificationType, setReplyNotificationType] = useState('any'); //todo: initialize with value from db/api
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const renderMentionSection = () => {
```

Comment:
this looks like it would be best to move to it's own component (file). Also it seems that the Text and FormattedText do not really apply any styles.

Reply:
done